### ISSUE TITLE
Conditional rendering in visual tests

### ISSUE THREAD

**Issue by averybross** [chromaui/chromatic-cli#526]
How can I show conditionally rendered elements in the visual tests?

or...

How can get visual tests of my components, AFTER the play function has run?

Is there a possible way to do this?

or do I need to create an HOC that handles the state/clickaway... and passes a open=true/false to the inner component?

So I can just write a story for the LOC and switch out the open prop?

Canvas
<img width="410" alt="image" src="https://user-images.githubusercontent.com/4257160/154721753-a14fabdd-9262-45fd-a02f-d0ca362bdd39.png">

Snapshot
<img width="389" alt="image" src="https://user-images.githubusercontent.com/4257160/154721777-94d905ef-9398-45ad-bd39-2bdbfa486569.png">

Open State code
```
Open.play = async ({ canvasElement }) => {
  const canvas = within(canvasElement)
  await userEvent.click(canvas.getByTestId('ButtonMenuCheckbox-button'))
  const menu = await waitFor(
    () => screen.getByTestId('ButtonMenuCheckbox-menu--loaded'),
    { timeout: 1000, interval: 2000 }
  )
  waitFor(() => screen.getByTestId('ButtonMenuCheckbox-menu--loaded'), {
    timeout: 2000,
    interval: 2000,
  })
  expect(menu).toBeTruthy()
}
```

**Note:** There is a 100ms opacity transition on the popper menu that I am trying to also get around. Any tips on this?

**Comment by kylesuss** [comment]
Hi @averybross -- as I understand it, play functions should work in Chromatic tests. Therefore, I would expect to see your menu. Let me ask though -- is your menu positioned absolutely? That is often the case which means that the menu is actually outside of the bounds of the `#root` element that is used for snapshotting. Oftentimes with elements positioned like this, you need to create a wrapper that takes up space & allows the menu to sit within the bounds of `#root`. Try that & let's see where you get!

**Comment by averybross** [comment]
Ah thanks!

**Comment by averybross** [comment]
Is there a way to set the height of root just for the tests?

**Comment by kylesuss** [comment]
Yeah that's a good method to try & see how it works. You might have to set the width too since the menu is wider than the button that opens it.

> Is there a way to set the height of root just for the tests?

You can detect whether or not you are running in Chromatic via this helper:

https://www.chromatic.com/docs/ischromatic

Most times though, this is only something you want on individual stories & not globally for all your stories so I wouldn't change `#root` for all stories, but you can certainly whip up a decorator for tooltip/menu stories like this that only runs in Chromatic.

**Comment by averybross** [comment]
Genius, thank you! Would there of been a better repo to report this issue on?

**Comment by averybross** [comment]
<img width="1731" alt="image" src="https://user-images.githubusercontent.com/4257160/154750304-995255c7-aa3f-40b6-ba52-682d734db564.png">

Looks like that did the trick, but now my float: right is not working in Chromatic... only in Storybook (which I will eventually fix using your helper)

**Comment by averybross** [comment]
Here it is in storybook

<img width="1410" alt="image" src="https://user-images.githubusercontent.com/4257160/154750559-dd3ad8fc-b971-484f-9fa0-d018b023b698.png">

**Comment by kylesuss** [comment]
Nice, glad it worked! Next time, you can use our in-app support widget (it is usually in the bottom right corner when you are logged in & using Chromatic) or just email support@chromatic.com.

We try to trim whitespace in certain scenarios which is likely why the float isn't showing up in the snapshot. I believe you can get around that by having 2 wrappers, 1 that sets the width to full & the other which is a descendent of that w/ the float.

**Comment by averybross** [comment]
Thanks!

**Comment by averybross** [comment]
Is there a fast way to produce these storyshots locally? Do I have to push and wait for all my GHA's to run before chromatic will produce a build?

**Comment by averybross** [comment]
@kylesuss Is there a fast way to produce these storyshots locally? Do I have to push and wait for all my GHA's to run before chromatic will produce a build?

**Comment by kylesuss** [comment]
You can run Chromatic builds locally. You'd basically be doing the exact same thing that the action is doing but calling the script manually:

https://www.chromatic.com/docs/cli#quick-start

**Comment by yannbf** [comment]
Hey @averybross I believe you can improve your play function to not need to use the `waitFor`:

```js
Open.play = async ({ canvasElement }) => {
  const canvas = within(canvasElement)
  await userEvent.click(canvas.getByTestId('ButtonMenuCheckbox-button'))
  // are you using screen because this is projected via portals outside of the story boundaries? like document.body 
  const menu = await screen.findByTestId('ButtonMenuCheckbox-menu--loaded')
  // testing-library matchers are available in @storybook/jest 0.0.9
  await expect(menu).toBeInTheDocument()
}
```
